Sources/Attribution/AttributionPoster.swift

```diff
@@ -167,9 +177,8 @@ final class AttributionPoster {
         let currentAppUserID = self.currentUserProvider.currentAppUserID
 
         // set the cache in advance to avoid multiple post calls
-        var newDictToCache = self.deviceCache.latestAdvertisingIdsByNetworkSent(appUserID: currentAppUserID)
-        newDictToCache[AttributionNetwork.adServices] = adServicesToken
-        self.deviceCache.set(latestAdvertisingIdsByNetworkSent: newDictToCache, appUserID: currentAppUserID)
+        var newDictToCache = self.subscriberAttributesManager.markAdServicesToken(adServicesToken,
+                                                                                  asSyncedFor: currentAppUserID)
```

Comment:
feels odd for subs attributes manager to be in charge for this - there's no subs attribute involved, right?

Reply:
I realize that this is piping through deviceCache anyway, but I feel like going directly might be more appropriate in this case - there's no subs attribute involved, but if I were reading this code with no context, I'd assume that the token is stored as a subs attribute

If we want to share the code, I think this could be either a part of DeviceCache itself or its own object, it just feels arbitrary to add it to subs attributes manager since it's not really related

Reply:
Hmm I agree. That's what I had originally, I _think_ I moved this so I could mock it, let me see.

Reply:
It's mainly because we don't mock `AttributionPoster`, and I wanted to verify this is called, but I'll see if I can do it in another way.

Reply:
Done